### PR TITLE
Added optional function to create text to show via callback

### PR DESCRIPTION
**Pull request by rvasqz86**

### CODE REVIEW

**Comment by MagTuxGit** [comment]
I need this "placeholder" option. Please, add this to the next version.

**Comment by Cretezy** [comment]
Rather use a generic `buildElement` function.

This is already currently supported, simply run `linkify` yourself and you can copy the `Linkify` widget.

File: lib/flutter_linkify.dart [hunk 1]
```diff
@@ -17,6 +17,9 @@ export 'package:linkify/linkify.dart'
 /// Callback clicked link
 typedef LinkCallback(LinkableElement link);
 
+/// Function for dynamic display text of link
+typedef String DisplayTextFn(String s);
```
Comment (Cretezy) [inline lib/flutter_linkify.dart:21]:
```suggestion
typedef String ElementFormatter(LinkifyElement element);
```

Let's make it more generic

File: lib/flutter_linkify.dart [hunk 2]
```diff
@@ -69,6 +72,9 @@ class Linkify extends StatelessWidget {
   /// Defines how to measure the width of the rendered text.
   final TextWidthBasis textWidthBasis;
 
+  /// Function for dynamic display text of link
+  final DisplayTextFn displayText;
```
Comment (Cretezy) [inline lib/flutter_linkify.dart:76]:
```suggestion
  final ElementFormatter formatter;
```

File: lib/flutter_linkify.dart [hunk 3]
```diff
@@ -278,13 +291,18 @@ TextSpan buildTextSpan(
   TextStyle style,
   TextStyle linkStyle,
   LinkCallback onOpen,
+  DisplayTextFn displayText,
```
Comment (Cretezy) [inline lib/flutter_linkify.dart:294]:
```suggestion
  ElementFormatter formatter,
```

File: lib/flutter_linkify.dart [hunk 4]
```diff
@@ -278,13 +291,18 @@ TextSpan buildTextSpan(
...
 }) {
   return TextSpan(
     children: elements.map<TextSpan>(
       (element) {
         if (element is LinkableElement) {
+          var text = element.text;
+          if(displayText != null) {
+            text = displayText(element.text);
+          }
```
Comment (Cretezy) [inline lib/flutter_linkify.dart:303]: Make `formatter` run on all `LinkifyElement`, not sure `LinkableElement`

**Comment by Cretezy** [comment]
Closing this as it has been active for several months.

**Pull request closed without merging**